src/ir/possible-contents.cpp

```diff
@@ -670,7 +670,16 @@ struct InfoCollector
     receiveChildValue(curr->ref, curr);
   }
   void visitRefAs(RefAs* curr) {
-    // TODO: optimize when possible: like RefCast, not all values flow through.
+    if (curr->op == ExternExternalize || curr->op == ExternInternalize) {
```

Comment:
Hmm, maybe these should be a separate expression class since they're materially different than the other `RefAs` cast instructions.

Reply:
Yeah, I think we considered that when you added those. Since then we've seen 2 places that needed special handling, so maybe it could have been separate I guess. Still if it's just those 2 places I think having less code is worthwhile.